### PR TITLE
Add Data Factory scanner

### PR DESCRIPTION
**Pull request by nachoalonsoportillo**
# Description

Adding Azure Data Factory support.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #103

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing

### CODE REVIEW

**Review by cmendible — CHANGES_REQUESTED** [review]
@nachoalonsoportillo thanks for this one! We need to add a couple of rules more and corresponding tests and we'll be ready to merge

File: internal/scanners/adf/rules.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+			Category:    scanners.RulesCategoryOperationalExcellence,
+			Subcategory: scanners.RulesSubcategoryOperationalExcellenceTags,
+			Description: "Azure Data Factory should have tags",
+			Severity:    scanners.SeverityLow,
+			Eval: func(target interface{}, scanContext *scanners.ScanContext) (bool, string) {
+				c := target.(*armdatafactory.Factory)
+				return len(c.Tags) == 0, ""
+			},
+			Url: "https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/tag-resources?tabs=json",
+		},
```
[the remark this answers is not in the document]
Comment (nachoalonsoportillo) [reply internal/scanners/adf/rules.go:64]: Completed. Done using the same approach you used in the AppServiceScanner (ie searching in the PrivateEndpoints map fetched from the ScanContext passed to the rule's function.)

[the remark this answers is not in the document]
Comment (nachoalonsoportillo) [reply internal/scanners/adf/rules.go:76]: Waiting for confirmation from Mario Zimmermann but, apparently, zonal protection is always there for the different components behind an instance of an ADF workspace. This is definitely not something that end users can configure.

[the remark this answers is not in the document]
Comment (nachoalonsoportillo) [reply internal/scanners/adf/rules.go:76]: It seems AZs is not there yet and is not expected to be before end of August. I wouldn't wait to merge this scanner until then or even later, when the GO SDK catches up with those changes should they show up on the REST API surface.

[the remark this answers is not in the document]
Comment (cmendible) [reply internal/scanners/adf/rules.go:76]: Agree

**Review by cmendible — APPROVED** [review]
LGTM

**Pull request closed without merging**